lib/api/src/lib.rs

```diff
@@ -28,6 +27,232 @@
+//!   Emscripten ABI.
+//! - [wasmer-wasi][] for running Wasm modules compiled to the WASI ABI.
+//!
+//! --------
+//!
+//! The Wasmer project has two major abstractions:
+//! - [Engines][wasmer-engine]
+//! - [Compilers][wasmer-compiler]
+//!
+//! These two abstractions have multiple options that can be enabled
+//! with features.
+//!
+//! ### Engines
+//!
+//! In the wasmer API, an engine is a system that uses a compiler to make
+//! a WebAssembly module executable.
+//!
+//! ### Compilers
+//!
+//! In the wasmer API, a compiler is a system that handles the details of
+//! making a Wasm module executable. For example, by generating native
+//! machine code for each Wasm function.
+//!
+//!
+//! ## Features
+//!
+//! This crate's features can be broken down into 2 kinds, features that
+//! enable new functionality and features that set defaults.
+//!
+//! The features that enable new functionality are:
+//! - `jit` - enable the JIT engine. (See [wasmer-jit][])
+//! - `native` - enable the native engine. (See [wasmer-native][])
+//! - `cranelift` - enable wasmer's Cranelift compiler. (See [wasmer-cranelift][])
+//! - `llvm` - enable wasmer's LLVM compiler. (See [wasmer-llvm][])
+//! - `singlepass` - enable wasmer's Singlepass compiler. (See [wasmer-singlepass][])
+//! - `wat` - enable wasmer to parse the WebAssembly text format.
+//!
+//! The features that set defaults come in sets that are mutually exclusive.
+//!
+//! The first set is the default compiler set:
+//! - `default-cranelift` - set wasmer's Cranelift compiler as the default.
+//! - `default-llvm` - set wasmer's LLVM compiler as the default.
+//! - `default-singlepass` - set wasmer's Singlepass compiler as the default.
+//!
+//! The next set is the default engine set:
+//! - `default-jit` - set the JIT engine as the default.
+//! - `default-native` - set the native engine as the default.
+//!
+//! --------
+//!
+//! By default the `wat`, `default-cranelift`, and `default-jit` features
+//! are enabled.
+//!
+//!
+//!
+//! [wasm]: https://webassembly.org/
+//! [wasmer-examples]: https://github.com/wasmerio/wasmer/tree/master/exmples
+//! [wasmer-cache]: https://docs.rs/wasmer-cache/1.0.0-beta1/wasmer_cache/
+//! [wasmer-compiler]: https://docs.rs/wasmer-compiler/1.0.0-beta1/wasmer_compiler/
+//! [wasmer-cranelift]: https://docs.rs/wasmer-cranelift/1.0.0-beta1/wasmer_cranelift/
+//! [wasmer-emscripten]: https://docs.rs/wasmer-emscripten/1.0.0-beta1/wasmer_emscripten/
+//! [wasmer-engine]: https://docs.rs/wasmer-engine/1.0.0-beta1/wasmer_engine/
+//! [wasmer-jit]: https://docs.rs/wasmer-jit/1.0.0-beta1/wasmer_jit/
+//! [wasmer-native]: https://docs.rs/wasmer-native/1.0.0-beta1/wasmer_native/
+//! [wasmer-singlepass]: https://docs.rs/wasmer-singlepass/1.0.0-beta1/wasmer_singlepass/
+//! [wasmer-llvm]: https://docs.rs/wasmer-llvm/1.0.0-beta1/wasmer_llvm/
+//! [wasmer-wasi]: https://docs.rs/wasmer-wasi/1.0.0-beta1/wasmer_wasi/
```

Comment:
Not sure I've edited all of them, but:
```suggestion
//! [`wasm]: https://webassembly.org/
//! [`wasmer-examples`]: https://github.com/wasmerio/wasmer/tree/master/exmples
//! [`wasmer-cache`]: https://docs.rs/wasmer-cache/*/wasmer_cache/
//! [`wasmer-compiler`]: https://docs.rs/wasmer-compiler/*/wasmer_compiler/
//! [`wasmer-cranelift`]: https://docs.rs/wasmer-cranelift/*/wasmer_cranelift/
//! [`wasmer-emscripten`]: https://docs.rs/wasmer-emscripten/*/wasmer_emscripten/
//! [`wasmer-engine`]: https://docs.rs/wasmer-engine/*/wasmer_engine/
//! [`wasmer-jit`]: https://docs.rs/wasmer-jit/*/wasmer_jit/
//! [`wasmer-native`]: https://docs.rs/wasmer-native/*/wasmer_native/
//! [` wasmer-singlepass`]: https://docs.rs/wasmer-singlepass/*/wasmer_singlepass/
//! [`wasmer-llvm`]: https://docs.rs/wasmer-llvm/*/wasmer_llvm/
//! [` wasmer-wasi`]: https://docs.rs/wasmer-wasi/*/wasmer_wasi/
```

For links, either we use `*` as a placeholder for the version to represent the latest version, but it implies that this crate uses the latest version of all those crates (which is very likely), or use specific version number but they are very likely to be outdated after some releases.

Thoughts?

Reply:
Nice, I didn't know about that, I think that's what we want